### ISSUE TITLE
'tuple' object has no attribute 'items' in models

### ISSUE THREAD

**Issue by pjbhaumik** [jdb78/pytorch-forecasting#1255]
- PyTorch-Forecasting version: 0.10.3
- PyTorch version: 1.10.2+cpu
- Python version: 3.8
- Operating System: jupyterlab notebook

### Expected behavior

I executed code Baseline().predict(val_dataloader) in order to get a baseline model  and expected to get result baseline predictions

### Actual behavior

However, result was 
![image](https://user-images.githubusercontent.com/48533013/218222023-4bb27202-1a41-4988-b1a2-27bae26a1f4e.png)

pytorch lightning returns a tuple from get_init_args(), so it won't have .items() attribute

### Code to reproduce the problem
training = TimeSeriesDataSet(
    df[lambda x: x.time_idx <= training_cutoff],
    time_idx="time_idx",
    target="NPD_LTR_nps",
    group_ids=["Label"],
    min_encoder_length=max_encoder_length // 2,  # keep encoder length long (as it is in the validation set)
    max_encoder_length=max_encoder_length,
    min_prediction_length=1,
    max_prediction_length=max_prediction_length,
    static_categoricals=["Label"],
    time_varying_known_reals=["time_idx"],
    time_varying_unknown_reals= df.drop(columns = ['time_idx', 'Label', 'FLT_LEG_SCHD_DPRT_LDT']).columns.to_list(),
    add_relative_time_idx=True,
    add_target_scales=True,
    add_encoder_length=True,
)

validation = TimeSeriesDataSet.from_dataset(training, df, predict=True, stop_randomization=True)

batch_size = 128  # set this between 32 to 128
train_dataloader = training.to_dataloader(train=True, batch_size=batch_size, num_workers=0)
val_dataloader = validation.to_dataloader(train=False, batch_size=batch_size * 10, num_workers=0)

actuals = torch.cat([y for x, (y, weight) in iter(val_dataloader)])
baseline_predictions = Baseline().predict(val_dataloader)
(actuals - baseline_predictions).abs().mean().item()

**Comment by Songloading** [comment]
I am getting same error from ```DeepAR.from_dataset```

**Comment by echolql** [comment]
I got same error from `**TemporalFusionTransformer.from_dataset**`

**Comment by bgrey001** [comment]
I am getting the same error for DeepAR.from_set and NBeats.from_dataset

**Comment by Da-byte** [comment]
I am getting the same error on Baseline, DeepAR, NHiTS and NBeats models. The error seems to have begun occurring 24h ago, whilst the same code worked perfectly 30h ago for me.

**Comment by robinkoppe** [comment]
I have the same error with TemporalFusionTransformer.from_dataset. 

PyTorch-Forecasting V 0.10.3
Python V 3.10.9
Torch V 1.13.1

**Comment by robinkoppe** [comment]
Seems like pytorch_lightning had an update yesterday (see link https://pypi.org/project/pytorch-lightning/#history). Reverting to version 1.9.0 worked for me. 

use `pip install pytorch_lightning==1.9.0`

**Comment by bgrey001** [comment]
> Seems like pytorch_lightning had an update yesterday (see link https://pypi.org/project/pytorch-lightning/#history). Reverting to version 1.9.0 worked for me.
> 
> use `pip install pytorch_lightning==1.9.0`

Can confirm this solves this issue seemingly, but I am faced with a myriad of other errors only trying to run the tutorial code. Has anyone been facing numerous bugs?

**Comment by anoetzli** [comment]
> 



> > Seems like pytorch_lightning had an update yesterday (see link https://pypi.org/project/pytorch-lightning/#history). Reverting to version 1.9.0 worked for me.
> > use `pip install pytorch_lightning==1.9.0`
> 
> Can confirm this solves this issue seemingly, but I am faced with a myriad of other errors only trying to run the tutorial code. Has anyone been facing numerous bugs?

Can confirm that downgrading pytorch_lightning to 1.9.0 resolved the issue, but I also get a myriad of other issues just running through the tutorial code, as the previous user also observed.

Yesterday I spent hours setting up different environments (conda, virtualenv) and attempted to setup an environment using different package managers (pip, conda), and could not proceed with the tutorials.

It would be helpful if someone could share a requirements file of an environment that actually works!

**Comment by pjbhaumik** [comment]
Hi, I am closing this issue, since the fix is to use the previous version of pytorch lightning per @robinkoppe.

However, I am facing several new issues as are others. Like @Da-byte, my code was working successfully before last Friday....


I am suggesting the use of a new module called darts (https://pypi.org/project/darts/) which implements TFT, DeepAR, NBEATS, NHiiTS and other methods such as AARIMA, ES etc...

**Comment by l-cdc** [comment]
This should be fixed by Lightning-AI/lightning#16851, which was included in pytorch-lightning 1.9.4 and 2.0.0. I could not test either yet but maybe that's useful.
